homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -26,10 +26,9 @@
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up the Islamic Prayer Component."""
     client = IslamicPrayerClient(hass, config_entry)
-
+    hass.data[DOMAIN] = client
```

Comment:
We shouldn't store things in hass.data before raising `ConfigEntryNotReady`. This integration needs to be refactored to split the platform setup from the client setup.